mcs/class/System/System.Net.NetworkInformation/NetworkInterface.cs

```diff
@@ -465,9 +465,13 @@ class Win32NetworkInterfaceAPI : NetworkInterfaceFactory
 			{
 				IntPtr ptr = IntPtr.Zero;
 				int len = 0;
-				GetAdaptersAddresses (0, 0, IntPtr.Zero, ptr, ref len);
+				uint flags = Win32_IP_ADAPTER_ADDRESSES.GAA_FLAG_INCLUDE_WINS_INFO | Win32_IP_ADAPTER_ADDRESSES.GAA_FLAG_INCLUDE_GATEWAYS;
+				GetAdaptersAddresses (0, flags, IntPtr.Zero, ptr, ref len);
+				if (Marshal.SizeOf (typeof (Win32_IP_ADAPTER_ADDRESSES)) > len)
```

Comment:
Are you suggesting checking the length on the Alignment structure in Win32_IP_Adapter_Addresses and comparing that against the size of the managed struct to make sure we are on a windows version that has all the fields we need? (I guess this is similar to the sanity checks above)

Reply:
Yes that was the idea (to check each converted item) since the above check just make sure there is at least one full Win32_IP_ADAPTER_ADDRESSES returned, but if there are multiple adapters returned (where each Win32_IP_ADAPTER_ADDRESSES is smaller than expected since we are on an earlier Windows version) it won't detect that condition. I did a check in corefx and it doesn't look like they do this additional checks though so maybe it is overkill. I'm fine leaving it as is.